library/src/main/java/com/novoda/downloadmanager/lib/DownloadService.java

```diff
@@ -370,9 +371,34 @@ private boolean updateLocked() {
             alarmManager.set(AlarmManager.RTC_WAKEUP, now + nextRetryTimeMillis, PendingIntent.getBroadcast(this, 0, intent, PendingIntent.FLAG_ONE_SHOT));
         }
 
+        if (!isActive) {
+            moveSubmittedTasksToBatchStatusIfNecessary();
+        }
+
         return isActive;
     }
 
+    private void moveSubmittedTasksToBatchStatusIfNecessary() {
```

Comment:
Yeah I guess that it will work, but I would prefer to leave my code in this PR and I encourage @ronocod to do that in another PR ;)